readthedocs/projects/views/private.py

```diff
@@ -506,19 +506,25 @@ def project_notifications(request, project_slug):
     project = get_object_or_404(
         Project.objects.for_admin_user(request.user), slug=project_slug)
 
-    email_form = EmailHookForm(data=request.POST or None, project=project)
-    webhook_form = WebHookForm(data=request.POST or None, project=project)
+    project_dashboard = reverse('projects_notifications', args=[project.slug])
 
-    if request.method == 'POST':
+    if request.method == 'POST' and 'email' in request.POST:
+        email_form = EmailHookForm(data=request.POST or None, project=project)
         if email_form.is_valid():
             email_form.save()
+            return HttpResponseRedirect(project_dashboard)
+    else:
+        # Blank email_form if webhook_form is submitted or the request is GET
+        email_form = EmailHookForm(data=None, project=project)
```

Comment:
I think this part isn't needed, would be better to use an `elif` here to check for `request.method == 'POST' and 'url' in request.POST:`

Reply:
looks like this is needed for the below request.

Reply:
Yup, it is required. @stsewd should I change it to use and elif like you said?

Reply:
Would be awesome to do something like that (for readability) but keeping the same logic. Anyway, is fine how it is now, I just didn't read the rest of the code when commenting that :grin:.